docs/static/azure-module.asciidoc

```diff
@@ -0,0 +1,451 @@
+  - ["insights-operational-logs", "Endpoint=sb://example1...", "DefaultEndpointsProtocol=https;AccountName=example1....", "HEAD",             "true"]
+  - ["insights-metrics-pt1m",     "Endpoint=sb://example2...", "DefaultEndpointsProtocol=https;AccountName=example2....", "TAIL",             "true"]
+-----
+
+[[azure_config_options]]
+===== Configuration options
+
+NOTE: All Event Hubs options are common to both basic and advanced
+configurations, with the following exceptions. The basic configuration uses
+`event-hub-connections`. The the advanced configuration uses `event_hubs` and
+`event_hub_connection`.
+
+[id="plugins-{type}s-{plugin}-event_hubs"]
+===== `event_hubs`
+* Value type is <<array,array>>
+* No default value
+* Ignored for basic configuration
+* Required for advanced configuration 
+
+Defines the Event Hubs to be read. An array of hashes where each entry is a
+hash of the Event Hub name and its configuration options.   
+
+
+[id="plugins-{type}s-{plugin}-event_hub_connections"]
+===== `event_hub_connections`
+* Value type is <<array,array>>
+* No default value
+* Required for basic configuration
+
+List of connection strings that identifies the Event Hubs to be read. Connection
+strings include the EntityPath for the Event Hub. 
+
+The `event_hub_connections` option is defined
+per Event Hub. All other configuration options are shared among Event Hubs.
+
+
+[id="plugins-{type}s-{plugin}-event_hub_connection"]
+===== `event_hub_connection`
+* Value type is <<string,string>>
+* No default value
+* Valid only for advanced configuration
+
+Connection string that identifies the Event Hub to be read. Advanced
+configuration options can be set per Event Hub. This option modifies
+`event_hub_name`, and should be nested under it. (See sample.) This option
+accepts only one connection string.
+
+
+[id="plugins-{type}s-{plugin}-checkpoint_interval"]
+===== `checkpoint_interval`
+* Value type is <<number,number>>
+* Default value is `5` seconds
+* Set to `0` to disable.
+  
+Interval in seconds to write checkpoints during batch processing. Checkpoints
+tell {ls} where to resume processing after a restart. Checkpoints are
+automatically written at the end of each batch, regardless of this setting. 
+
+Writing checkpoints too frequently can slow down processing unnecessarily. 
+
+
+[id="plugins-{type}s-{plugin}-consumer_group"]
+===== `consumer_group`
+* Value type is <<string,string>>
+* Default value is `$Default` 
+
+Consumer group used to read the Event Hub(s). Create a consumer group
+specifically for Logstash. Then ensure that all instances of Logstash use that
+consumer group so that they can work together properly.
+
+
+[id="plugins-{type}s-{plugin}-decorate_events"]
+===== `decorate_events`
+
+* Value type is <<boolean,boolean>>
+* Default value is `false`
+
+Adds metadata about the Event Hub, including Event Hub name, consumer_group,
```

Comment:
Should these be formatted in backticks? If not, maybe change this to say consumer group, processor host, and so on (without underscores).